File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominPrivacyConfig.java
package ua.gomin.messenger.configs;

import android.content.Context;
import android.content.SharedPreferences;

public class GominPrivacyConfig {

    public static final GominPrivacyConfig INSTANCE = new GominPrivacyConfig();
    private static final String PREFS_NAME = "gomin_privacy_config";

    

    private SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    public boolean getGhostModeReadMessages(Context context) {
        return getPrefs(context).getBoolean("ghostModeReadMessages", false);
    }
    public boolean getGhostModeHideTyping(Context context) {
        return getPrefs(context).getBoolean("ghostModeHideTyping", false);
    }
    public boolean getGhostModeHideStoryViews(Context context) {
        return getPrefs(context).getBoolean("ghostModeHideStoryViews", false);
    }
    public boolean getGhostModeHideOnline(Context context) {
        return getPrefs(context).getBoolean("ghostModeHideOnline", false);
    }
    public boolean getKeepDeletedMessages(Context context) {
        return getPrefs(context).getBoolean("keepDeletedMessages", false);
    }
    public boolean getAskBiometricsToOpenChat(Context context) {
        return getPrefs(context).getBoolean("askBiometricsToOpenChat", false);
    }
    public boolean getAskBiometricsToOpenEncrypted(Context context) {
        return getPrefs(context).getBoolean("askBiometricsToOpenEncrypted", false);
    }
    public boolean getAskBiometricsToOpenArchive(Context context) {
        return getPrefs(context).getBoolean("askBiometricsToOpenArchive", false);
    }
    public boolean getAskPasscodeBeforeDelete(Context context) {
        return getPrefs(context).getBoolean("askPasscodeBeforeDelete", false);
    }
    public boolean getAllowSystemPasscode(Context context) {
        return getPrefs(context).getBoolean("allowSystemPasscode", false);
    }
    public void setGhostModeReadMessages(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("ghostModeReadMessages", value).apply();
    }
    public void setGhostModeHideTyping(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("ghostModeHideTyping", value).apply();
    }
    public void setGhostModeHideStoryViews(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("ghostModeHideStoryViews", value).apply();
    }
    public void setGhostModeHideOnline(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("ghostModeHideOnline", value).apply();
    }
    public void setKeepDeletedMessages(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("keepDeletedMessages", value).apply();
    }
    public void setAskBiometricsToOpenChat(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenChat", value).apply();
    }
    public void setAskBiometricsToOpenEncrypted(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenEncrypted", value).apply();
    }
    public void setAskBiometricsToOpenArchive(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenArchive", value).apply();
    }
    public void setAskPasscodeBeforeDelete(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("askPasscodeBeforeDelete", value).apply();
    }
    public void setAllowSystemPasscode(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("allowSystemPasscode", value).apply();
    }
}

File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominPrivacyConfig.kt
package ua.gomin.messenger.configs

import android.content.Context
import android.content.SharedPreferences

/**
 * Налаштування приватності та Ghost Mode.
 */
object GominPrivacyConfig {

    private const val PREFS_NAME = "gomin_privacy_config"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    // Ghost Mode
    fun getGhostModeReadMessages(context: Context): Boolean =
        getPrefs(context).getBoolean("ghostModeReadMessages", false)

    fun setGhostModeReadMessages(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("ghostModeReadMessages", enabled).apply()
    }

    fun getGhostModeHideTyping(context: Context): Boolean =
        getPrefs(context).getBoolean("ghostModeHideTyping", false)

    fun setGhostModeHideTyping(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("ghostModeHideTyping", enabled).apply()
    }

    fun getGhostModeHideStoryViews(context: Context): Boolean =
        getPrefs(context).getBoolean("ghostModeHideStoryViews", false)

    fun setGhostModeHideStoryViews(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("ghostModeHideStoryViews", enabled).apply()
    }

    fun getGhostModeHideOnline(context: Context): Boolean =
        getPrefs(context).getBoolean("ghostModeHideOnline", false)

    fun setGhostModeHideOnline(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("ghostModeHideOnline", enabled).apply()
    }

    // Anti-delete
    fun getKeepDeletedMessages(context: Context): Boolean =
        getPrefs(context).getBoolean("keepDeletedMessages", false)

    fun setKeepDeletedMessages(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("keepDeletedMessages", enabled).apply()
    }

    // Biometric security
    fun getAskBiometricsToOpenChat(context: Context): Boolean =
        getPrefs(context).getBoolean("askBiometricsToOpenChat", false)

    fun setAskBiometricsToOpenChat(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenChat", enabled).apply()
    }

    fun getAskBiometricsToOpenEncrypted(context: Context): Boolean =
        getPrefs(context).getBoolean("askBiometricsToOpenEncrypted", false)

    fun setAskBiometricsToOpenEncrypted(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenEncrypted", enabled).apply()
    }

    fun getAskBiometricsToOpenArchive(context: Context): Boolean =
        getPrefs(context).getBoolean("askBiometricsToOpenArchive", false)

    fun setAskBiometricsToOpenArchive(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("askBiometricsToOpenArchive", enabled).apply()
    }

    fun getAskPasscodeBeforeDelete(context: Context): Boolean =
        getPrefs(context).getBoolean("askPasscodeBeforeDelete", false)

    fun setAskPasscodeBeforeDelete(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("askPasscodeBeforeDelete", enabled).apply()
    }

    fun getAllowSystemPasscode(context: Context): Boolean =
        getPrefs(context).getBoolean("allowSystemPasscode", false)

    fun setAllowSystemPasscode(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("allowSystemPasscode", enabled).apply()
    }
}
